tesseract/tesseract_common/cmake/tesseract_macros.cmake

```diff
@@ -41,11 +45,33 @@ macro(tesseract_variables)
   endif()
 
   if (NOT TESSERACT_ENABLE_TESTING AND NOT TESSERACT_ENABLE_TESTING_ALL)
-    set(TESSERACT_COMPILE_OPTIONS -Wall -Wextra -Wconversion -Wsign-conversion -Wno-sign-compare)
     set(TESSERACT_CLANG_TIDY_ARGS "-header-filter=.*" "-line-filter=[{'name':'EnvironmentMonitorDynamicReconfigureConfig.h','lines':[[9999999,9999999]]}, {'name':'.h'}, {'name':'.hpp'}]" "-checks=-*,clang-analyzer-*,bugprone-*,cppcoreguidelines-avoid-goto,cppcoreguidelines-c-copy-assignment-signature,cppcoreguidelines-interfaces-global-init,cppcoreguidelines-narrowing-conversions,cppcoreguidelines-no-malloc,cppcoreguidelines-slicing,cppcoreguidelines-special-member-functions,misc-*,modernize-*,performance-*,readability-avoid-const-params-in-decls,readability-container-size-empty,readability-delete-null-pointer,readability-deleted-default,readability-else-after-return,readability-function-size,readability-identifier-naming,readability-inconsistent-declaration-parameter-name,readability-misleading-indentation,readability-misplaced-array-index,readability-non-const-parameter,readability-redundant-*,readability-simplify-*,readability-static-*,readability-string-compare,readability-uniqueptr-delete-release,readability-rary-objects")
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      set(TESSERACT_COMPILE_OPTIONS -Wall -Wextra -Wconversion -Wsign-conversion -Wno-sign-compare)
+      set(TESSERACT_COMPILE_DEFINITIONS TESSERACT_BUILDING_LIBRARY=ON)
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+      set(TESSERACT_COMPILE_OPTIONS -Wall -Wextra -Wconversion -Wsign-conversion)
+      set(TESSERACT_COMPILE_DEFINITIONS TESSERACT_BUILDING_LIBRARY=ON)
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+      set(TESSERACT_COMPILE_OPTIONS "")
+      set(TESSERACT_COMPILE_DEFINITIONS _USE_MATH_DEFINES=ON TESSERACT_BUILDING_LIBRARY=ON)
+    else()
+      message(WARNING "${CMAKE_CXX_COMPILER_ID} Unsupported compiler detected.")
+    endif()
```

Comment:
Perhaps some comments here describing the decisions here. For example, why doesn't clang check for sign-compare?

Reply:
I believe it is just not available on clang, but I will check.

Reply:
Yea clang currently does not have a sign-compare option.